java/src/main/java/com/google/api/services/samples/youtube/cmdline/live/ListLiveChatMessages.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.api.services.samples.youtube.cmdline.live;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.services.samples.youtube.cmdline.Auth;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.YouTubeScopes;
+import com.google.api.services.youtube.model.LiveChatMessage;
+import com.google.api.services.youtube.model.LiveChatMessageAuthorDetails;
+import com.google.api.services.youtube.model.LiveChatMessageListResponse;
+import com.google.api.services.youtube.model.LiveChatMessageSnippet;
+import com.google.api.services.youtube.model.LiveChatSuperChatDetails;
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * Lists live chat messages and SuperChat details from a live broadcast.
+ *
+ * The videoId is often included in the video's url, e.g.:
+ * https://www.youtube.com/watch?v=L5Xc93_ZL60
+ *                                 ^ videoId
+ * The video URL may be found in the browser address bar, or by right-clicking a video and selecting
+ * Copy video URL from the context menu.
+ *
+ * @author Jim Rogers
+ */
+public class ListLiveChatMessages {
+
+    /**
+     * Common fields to retrieve for chat messages
+     */
+    private static final String LIVE_CHAT_FIELDS =
+        "items(authorDetails(channelId,displayName,isChatModerator,isChatOwner,isChatSponsor,"
+            + "profileImageUrl),snippet(displayMessage,superChatDetails,publishedAt)),"
+            + "nextPageToken,pollingIntervalMillis";
```

Comment:
True, though that assumes that you're storing that ID with each insert. A moderator might insert multiple messages. 

Again, though, no need to modify at this point.